addOns/graphql/src/main/java/org/zaproxy/addon/graphql/ImportDialog.java

```diff
@@ -0,0 +1,245 @@
+                                Constant.messages.getString(
+                                        "graphql.error.invalidurl", e.getMessage()));
+                        fieldEndpoint.requestFocusInWindow();
+                    });
+        }
+        return false;
+    }
+
+    private JButton getChooseFileButton() {
+        if (buttonChooseFile == null) {
+            buttonChooseFile =
+                    new JButton(Constant.messages.getString(MESSAGE_PREFIX + "chooseFileButton"));
+            buttonChooseFile.addActionListener(
+                    e -> {
+                        JFileChooser filechooser =
+                                new JFileChooser(
+                                        Model.getSingleton().getOptionsParam().getUserDirectory());
+                        int state = filechooser.showOpenDialog(View.getSingleton().getMainFrame());
+                        if (state == JFileChooser.APPROVE_OPTION) {
+                            String filename = filechooser.getSelectedFile().getAbsolutePath();
+                            try {
+                                getSchemaField().setText(filename);
+                                Model.getSingleton()
+                                        .getOptionsParam()
+                                        .setUserDirectory(filechooser.getCurrentDirectory());
+                            } catch (Exception e1) {
+                                showWarningDialog(
+                                        Constant.messages.getString(
+                                                "graphql.error.filenotfound", filename));
+                            }
+                        }
+                    });
+        }
+        return buttonChooseFile;
+    }
+
+    public boolean importDefinition() {
+        String schemaLocation = getSchemaField().getText();
+        try {
+            if (schemaLocation == null || schemaLocation.isEmpty()) {
+                getParser().introspect();
+            } else if (UrlBuilder.isValidUrl(schemaLocation)) {
+                getParser().importUrl(schemaLocation);
+            } else {
+                getParser().importFile(schemaLocation);
+            }
+            return true;
+        } catch (Exception e) {
+            ThreadUtils.invokeAndWaitHandled(
+                    () -> {
+                        showWarningDialog(
+                                Constant.messages.getString(
+                                        "graphql.error.import", e.getMessage()));
+                        if (schemaLocation == null || schemaLocation.isEmpty()) {
+                            fieldEndpoint.requestFocusInWindow();
+                        } else {
+                            fieldSchema.requestFocusInWindow();
+                        }
+                    });
+        }
+        return false;
+    }
+
+    protected void showWarningDialog(String message) {
+        View.getSingleton().showWarningDialog(this, message);
```

Comment:
It would be better that the dialogue was made visible here, as is, the warn dialogue is shown misplaced and sometimes hidden behind the main ZAP window (since this dialogue is not visible it's not properly placed).